api/src/main/java/org/apache/iceberg/rest/CreateNamespaceRequest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.io.Serializable;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+/**
+ * Represents a REST request to create a namespace / database.
+ */
+public class CreateNamespaceRequest implements Serializable {
+
+  // TODO - Use protected so users can extend this for their own impls. Or an interface.
```

Comment:
Are user implementations to be supported? I was thinking that we would treat the REST API as a spec so it wouldn't necessarily make sense to allow customization.

Reply:
You're right. I actually meant more what you said (about it as a spec). Bad phrasing on my part. I have been working through various versions and different code pathways, and ultimately decided to introduce this initially with `CreateNamespaceRequest` to open discussion.

I'll update this to be more clear that this designed to be a spec.